google/cloud/spanner/internal/connection_impl_test.cc

```diff
@@ -408,6 +408,73 @@ TEST(ConnectionImplTest, ReadImplicitBeginTransaction) {
   EXPECT_THAT(txn, HasSessionAndTransactionId("test-session-name", "ABCDEF00"));
 }
 
+TEST(ConnectionImplTest, ReadImplicitBeginTransactionOneTransientFailure) {
+  auto mock = std::make_shared<spanner_testing::MockSpannerStub>();
+  auto db =
+      spanner::Database("dummy_project", "dummy_instance", "dummy_database_id");
+  auto conn = MakeConnection(
+      db, {mock},
+      spanner::ConnectionOptions{grpc::InsecureChannelCredentials()});
+  grpc::Status grpc_status(grpc::StatusCode::PERMISSION_DENIED, "uh-oh");
```

Comment:
`PERMISSION_DENIED` is not usually transient, did I miss something?

Reply:
If an RPC with a piggy-backed "begin transaction" fails for any reason, we do an explicit BeginTransaction() and then (if that succeeded) try the RPC again.  We need to do that to maintain the transaction semantics.